src/cargo/core/compiler/build_context/mod.rs

```diff
@@ -61,7 +61,10 @@ impl<'a, 'cfg> BuildContext<'a, 'cfg> {
         units: &'a UnitInterner<'a>,
         extra_compiler_args: HashMap<Unit<'a>, Vec<String>>,
     ) -> CargoResult<BuildContext<'a, 'cfg>> {
-        let rustc = config.load_global_rustc(Some(ws))?;
+        let mut rustc = config.load_global_rustc(Some(ws))?;
+        if let Some(wrapper) = &build_config.rustc_wrapper {
```

Comment:
I would not add this, and leave `BuildConfig.primary_unit_rustc` working the way it was before.

Reply:
The usges of primary unit rustc got moved to be sourcing from the member in the Rustc struct which is why this got modified, but I agree this needs to be changed because it's currently just a complete override of the wrapper for all crates